.github/workflows/scripts/build.py

```diff
@@ -1,25 +1,75 @@
 #!/usr/bin/env python3
 
-# -- prioritized --
 import ci_common  # isort: skip, early initialization happens here
 
-# -- stdlib --
 import glob
 import os
 import platform
+from pathlib import Path
 
-# -- third party --
-# -- own --
 from ci_common.dep import download_dep
-from ci_common.misc import banner, get_cache_home, is_manylinux2014
-from ci_common.python import setup_python
+from ci_common.misc import (banner, get_cache_home, is_manylinux2014,
+                            path_prepend)
+from ci_common.python import path_prepend, setup_python
 from ci_common.sccache import setup_sccache
-from ci_common.tinysh import Command, environ, git, sh
+from ci_common.tinysh import Command, git, sh
 
 
 # -- code --
+@banner('Setup Clang')
+def setup_clang(as_compiler=True) -> None:
+    '''
+    Setup Clang.
+    '''
+    u = platform.uname()
+    if u.system == 'Linux':
+        pass
+    elif (u.system, u.machine) == ('Windows', 'AMD64'):
+        out = get_cache_home() / 'clang-15'
+        url = 'https://github.com/python3kgae/taichi_assets/releases/download/llvm15_vs2022_clang/clang-15.0.0-win.zip'
+        download_dep(url, out)
+        path_prepend('PATH', out / 'bin')
+        os.environ['TAICHI_CMAKE_ARGS'] += ' -DCLANG_EXECUTABLE=clang++.exe'
+
+        if as_compiler:
+            os.environ['TAICHI_CMAKE_ARGS'] += (' -DCMAKE_CXX_COMPILER=clang++'
```

Comment:
This flag might not do anything under MSBuild as it uses its own "Toolset" system

Reply:
(Today I learned!

The flag will not be set when using MSVC (guarded by `as_compiler`, ). See https://github.com/taichi-dev/taichi/pull/6993/files#diff-08e92060a244d51600d95a8a4e32047aa4f3c4bbb856becf14d67c83b6d12907R183